packages/api/src/error-handler.js

```diff
@@ -6,18 +6,23 @@ import { HTTPError, PinningServiceApiError } from './errors.js'
  * @param {Error & {status?: number;code?: string;reason?: string; details?: string; }} err
  * @param {import('./env').Env} env
  */
-export function errorHandler (err, { log }) {
+export function errorHandler (err, { log }, request) {
   console.error(err.stack)
 
   let status = err.status || 500
   if (status >= 500) {
     log.error(err)
   }
 
-  if (err instanceof PinningServiceApiError) {
+  // As the pinning service requires a certain specification
+  // Always return the nested error object for any pinning service endpoints
+  const requestUrl = new URL(request.url)
+  if (err instanceof PinningServiceApiError || (requestUrl && /^\/pins/.test(requestUrl.pathname))) {
```

Comment:
I see you're checking `requestUrl &&` 
Can requestUrl be `undefined` here? Wouldn't `new URL` throw for invalid URLs?

Reply:
Good point. Removed this check.